### PR TITLE
Asynchronize some blocking waits

### PR DESCRIPTION
**Pull request by Therzok**
/cc @mkrueger for https://github.com/mono/monodevelop/commit/0e3860e8096ef904102b98fa35b3fd7a3ca5eaf8 and https://github.com/mono/monodevelop/commit/d675b4a4b6d2eabce39e473f8765762559ade5e1
/cc @slluis for https://github.com/mono/monodevelop/commit/031d1ba709192062f07abd696cf4d0a842c126e3

### CODE REVIEW

**Comment by slluis** [comment]
Ok from my side

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp.Parser/CSharpParsedDocument.cs [hunk 1]
```diff
@@ -435,24 +435,26 @@ public override Task<IReadOnlyList<Error>> GetErrorsAsync (CancellationToken can
...
-								errors = emptyErrors;
-							} catch (Exception e) {
-								LoggingService.LogError ("Error while getting diagnostics.", e);
-								errors = emptyErrors;
-							}
-						}
-					}
-					return errors;
+				return Task.Run (async delegate {
+					await errorLock.WaitAsync ();
```
Comment (alanmcgovern) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Parser/CSharpParsedDocument.cs]: Maybe use the overload which accepts the `cancellationToken` :)

Comment (Therzok) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp.Parser/CSharpParsedDocument.cs]: 👍

**Comment by Therzok** [comment]
Merged manually.

**Pull request closed without merging**